cluster-autoscaler/simulator/node_info_utils.go

```diff
@@ -58,33 +62,44 @@ func SanitizedTemplateNodeInfoFromNodeInfo(example *framework.NodeInfo, nodeGrou
 
 	// We need to sanitize the example before determining the DS pods, since taints are checked there, and
 	// we might need to filter some out during sanitization.
-	sanitizedExample := sanitizeNodeInfo(example, newNodeNameBase, randSuffix, &taintConfig)
+	sanitizedExample, err := sanitizeNodeInfo(example, newNodeNameBase, randSuffix, &taintConfig)
+	if err != nil {
+		return nil, errors.ToAutoscalerError(errors.InternalError, err)
+	}
 	expectedPods, err := podsExpectedOnFreshNode(sanitizedExample, daemonsets, forceDaemonSets, randSuffix)
 	if err != nil {
-		return nil, err
+		return nil, errors.ToAutoscalerError(errors.InternalError, err)
 	}
 	// No need to sanitize the expected pods again - they either come from sanitizedExample and were sanitized above,
 	// or were added by podsExpectedOnFreshNode and sanitized there.
-	return framework.NewNodeInfo(sanitizedExample.Node(), nil, expectedPods...), nil
+	return framework.NewNodeInfo(sanitizedExample.Node(), sanitizedExample.LocalResourceSlices, expectedPods...), nil
 }
 
 // NodeInfoSanitizedDeepCopy duplicates the provided template NodeInfo, returning a fresh NodeInfo that can be injected into the cluster snapshot.
 // The NodeInfo is sanitized (names, UIDs are changed, etc.), so that it can be injected along other copies created from the same template.
-func NodeInfoSanitizedDeepCopy(template *framework.NodeInfo, suffix string) *framework.NodeInfo {
+func NodeInfoSanitizedDeepCopy(template *framework.NodeInfo, suffix string) (*framework.NodeInfo, error) {
 	// Template node infos should already have taints and pods filtered, so not setting these parameters.
 	return sanitizeNodeInfo(template, template.Node().Name, suffix, nil)
 }
 
-func sanitizeNodeInfo(nodeInfo *framework.NodeInfo, newNodeNameBase string, namesSuffix string, taintConfig *taints.TaintConfig) *framework.NodeInfo {
+func sanitizeNodeInfo(nodeInfo *framework.NodeInfo, newNodeNameBase string, namesSuffix string, taintConfig *taints.TaintConfig) (*framework.NodeInfo, error) {
 	freshNodeName := fmt.Sprintf("%s-%s", newNodeNameBase, namesSuffix)
 	freshNode := sanitizeNode(nodeInfo.Node(), freshNodeName, taintConfig)
-	result := framework.NewNodeInfo(freshNode, nil)
+	freshResourceSlices, oldPoolNames, err := drautils.SanitizeNodeResourceSlices(nodeInfo.LocalResourceSlices, freshNode.Name, namesSuffix)
+	if err != nil {
+		return nil, err
+	}
+	result := framework.NewNodeInfo(freshNode, freshResourceSlices)
 
 	for _, podInfo := range nodeInfo.Pods() {
 		freshPod := sanitizePod(podInfo.Pod, freshNode.Name, namesSuffix)
-		result.AddPod(framework.NewPodInfo(freshPod, nil))
+		freshResourceClaims, err := drautils.SanitizePodResourceClaims(freshPod, podInfo.Pod, podInfo.NeededResourceClaims, namesSuffix, nodeInfo.Node().Name, freshNodeName, oldPoolNames)
```

Comment:
It's not immediately obvious that `drautils.SanitizePodResourceClaims` doesn't change something in place. Perhaps rename it to `drautils.SanitizePodResourceClaimsDeepCopy` for consistency with this `NodeInfoSanitizedDeepCopy` here?

Or is it a convention that `sanitize<something>` returns a deep copy? `sanitizePod` and `sanitizeNode` below do that. Then why the `DeepCopy` suffix in `NodeInfoSanitizedDeepCopy`?

Reply:
+1, it seems like all of the `sanitize.go` funcs produce a new copy. Would the following names be clearer?

- `GetSanitizedNodeResourceSlices`
- `GetSanitizedPodResourceClaims`
- `GetSanitizedResourceClaimRefs`

Or maybe follow the pattern from this package, e.g. `SanitizedTemplateNodeInfoFromNodeGroup` (note the "d" in Sanitized...) I probably prefer the clarity of the `Get` prefix but I understand why that might not be needed.

And probably `NodeInfoSanitizedDeepCopy` can be changed to either `SanitizedNodeInfo` or `GetSanitizedNodeInfo` (depending on what pattern we choose to standardize to).

Reply:
Good catch, it's indeed a bit of convention that the `Sanitized<something>` functions return a deep-copy, but as Jack noted the "d" is important here - my bad. Standardized the names following Jacks suggestions:

* `SanitizeNodeResourceSlices` -> `SanitizedNodeResourceSlices` (without the Get prefix to match Go conventions, as it's an exported function)
* `SanitizePodResourceClaims` -> `SanitizedPodResourceClaims` (same reasoning as above)
* `SanitizeResourceClaimRefs` -> `SanitizedResourceClaimRefs` (same reasoning as above)
* `NodeInfoSanitizedDeepCopy` -> `SanitizedNodeInfo` (same reasoning as above)
* `sanitizeNodeInfo` -> `createSanitizedNodeInfo` (added the "create" prefix here as it's a local function and `sanitizedNodeInfo` is used as a variable name, "create" fits better than "get" IMO).
* `sanitizeNode` -> `createSanitizedNode` (same reasoning as above).
* `sanitizePod` -> `createSanitizedPod` (same reasoning as above).